pkg/env/msiauthorizer.go

```diff
@@ -20,14 +20,22 @@ const (
 	MSIContextGateway MSIContext = "GATEWAY"
 )
 
-func (c *core) NewMSITokenCredential(msiContext MSIContext) (azcore.TokenCredential, error) {
+func (c *core) NewMSITokenCredential() (azcore.TokenCredential, error) {
 	if !c.IsLocalDevelopmentMode() {
 		options := c.Environment().ManagedIdentityCredentialOptions()
 		return azidentity.NewManagedIdentityCredential(options)
 	}
+
+	var msiContext string
+	if c.component == COMPONENT_GATEWAY {
+		msiContext = string(MSIContextGateway)
+	} else {
+		msiContext = string(MSIContextRP)
+	}
```

Comment:
This feels a little prone to failure later on in weird ways. If we want envs to be our source of truth, would it be better to put this all in objects there? So we get a `component` type with `name` and `msiContext` attributes? That way we're forced to always define what context each component gets, and it's immediately obvious in a single location?

Reply:
Yeah, this is a good idea for further work on it, especially as MIMO will have its own component contexts and permissions.